taker/src/lib.rs

```diff
@@ -0,0 +1,539 @@
+
+    /// If enabled logs will be in json format
+    #[clap(short, long)]
+    json: bool,
+
+    /// If enabled, logs in json format will contain a list of all ancestor spans of log events.
+    /// This **only** has an effect when `json` is also enabled.
+    #[clap(long)]
+    pub json_span_list: bool,
+
+    /// If enabled, traces will be exported to the OTEL collector
+    #[clap(long)]
+    instrumentation: bool,
+
+    /// If enabled, tokio runtime can be locally debugged with tokio_console
+    #[clap(long)]
+    pub tokio_console: bool,
+
+    /// If enabled, libp2p ping and offer broadcast spans will be included in the traces exported
+    /// by the application.
+    #[clap(long)]
+    pub verbose_spans: bool,
+
+    /// OTEL collector endpoint address
+    ///
+    /// If not specified it defaults to the local collector endpoint.
+    #[clap(long, default_value = LOCAL_COLLECTOR_ENDPOINT )]
+    collector_endpoint: String,
+
+    /// If enabled, browser UI is not automatically launched at startup.
+    #[clap(long)]
+    pub headless: bool,
+
+    /// Service name for OTEL.
+    ///
+    /// If not specified it defaults to the binary name.
+    #[clap(long, default_value = "taker")]
+    service_name: String,
+
+    /// Configure the log level, e.g.: one of Error, Warn, Info, Debug, Trace
+    #[clap(short, long, default_value = "Debug")]
+    log_level: LevelFilter,
+
+    /// Password for the web interface.
+    ///
+    /// If not provided, will be derived from the seed.
+    #[clap(long)]
+    password: Option<Password>,
+
+    #[clap(subcommand)]
+    network: Option<Network>,
+
+    #[clap(short, long, parse(try_from_str = parse_app_seed))]
+    app_seed: Option<[u8; 32]>,
+
+    /// If provided will be used for internal wallet instead of a random key or app_seed. The
+    /// keys will be derived according to Bip84.
+    #[clap(short, long)]
+    pub wallet_xprv: Option<ExtendedPrivKey>,
+}
+
+impl Opts {
+    pub fn read() -> Opts {
+        Opts::parse()
+    }
+
+    pub fn new(network: String, password: String, data_dir: String) -> Opts {
+        // mainnet is the default setting nothing to do if network is mainnet.
+        let mut opts = Opts::default();
+
+        // signet, regtest are not implemented, since there is no maker available.
+        if network == "testnet" {
+            opts.maker = Some(TESTNET_MAKER.to_string());
+            opts.maker_id = Some(parse_x25519_pubkey(TESTNET_MAKER_ID).expect("public key"));
+            opts.maker_peer_id =
+                Some(PeerId::from_str(TESTNET_MAKER_PEER_ID).expect("maker peer id"));
+            opts.network = Some(Network::Testnet {
+                electrum: "ssl://blockstream.info:993".to_string(),
+                withdraw: None,
+            });
+        }
+
+        opts.data_dir = Some(PathBuf::from(data_dir));
+        if !password.is_empty() {
+            opts.password = Some(Password::from_str(&password).expect("password"));
```

Comment:
🙃
I think this deserve a better error message then `password` :)

Reply:
since we have now an acceptable default + change password, I removed the password option all together.